sdk/core/Azure.Core.Experimental/src/DynamicData.cs

```diff
@@ -1,28 +1,202 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections;
+using System.Diagnostics;
 using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core.Json;
 
 namespace Azure.Core.Dynamic
 {
     /// <summary>
-    /// A dynamic abstraction over content data.  Deriving types are implemented
-    /// for a specific format, such as JSON or XML.
-    ///
-    /// This and related types are not intended to be mocked.
+    /// Dynamic layer over MutableJsonDocument.
     /// </summary>
-    public abstract class DynamicData
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [JsonConverter(typeof(JsonConverter))]
```

Comment:
@KrzysztofCwalina, would we have any concerns with this type being JSON-serializable, in any of our extensibility scenarios?  If it were holding e.g. XML, I would expect JSON serialization to throw, but it doesn't feel like it precludes supporting serialization to other formats.

Thoughts?